uvicorn/main.py

```diff
@@ -428,6 +428,9 @@ def run(self, sockets=None):
             except OSError as exc:
                 logger.error(exc)
                 sys.exit(1)
+            port = config.port
+            if port == 0:
+                port = server.sockets[0].getsockname()[1]
```

Comment:
Will `server.sockets` only ever have a single item in the list here? I think so but I'm not certain.